### PR TITLE
Crawl tool

### PR DESCRIPTION
**Pull request by nhatminh-96**
Add script to crawl publications and reports from ECMWF homepage.

### CODE REVIEW

File: crawl_tool/README.md [hunk 1]
```diff
@@ -0,0 +1,17 @@
+# Crawl tool for ECMWF homepage
+This script will download publications and reports from ECMWF homepage under pdf format.
+
+## How to use
+
+Run the script crawl.py: 
+`python crawl.py --source memoranda --pages [1,10] --download`
```
Comment (thinhngo-x) [inline crawl_tool/README.md:7]: may need to add `cd crawl_tool` before

File: crawl_tool/crawl.py [hunk 2]
```diff
@@ -0,0 +1,158 @@
+"""Crawl publications from ECMWF homepage."""
+
+import os
+import re
+import time
+import urllib.request
+
+import click
```
Comment (thinhngo-x) [inline crawl_tool/crawl.py:8]: a small question, why use `click` instead of the common `argparse`

File: crawl_tool/crawl.py [hunk 3]
```diff
@@ -0,0 +1,158 @@
...
+from bs4 import BeautifulSoup
+
+OUTPUT_ROOT = "/home/user/large-disk/crawled_resources/"
```
Comment (thinhngo-x) [inline crawl_tool/crawl.py:11]: `output_root` should be set as an argument

File: crawl_tool/crawl.py (hunk 2 above)
Comment (nhatminh-96) [reply crawl_tool/crawl.py:8]: https://click.palletsprojects.com/en/8.1.x/why/#why-not-argparse
- argparse has built-in behavior to guess if something is an argument or an option. This becomes a problem when dealing with incomplete command lines; the behaviour becomes unpredictable without full knowledge of a command line. This goes against Click’s ambitions of dispatching to subparsers.
- argparse does not support disabling interspersed arguments. Without this feature, it’s not possible to safely implement Click’s nested parsing.

File: crawl_tool/README.md (hunk 1 above)
Comment (nhatminh-96) [reply crawl_tool/README.md:7]: Okay to be more specific.

File: crawl_tool/crawl.py (hunk 3 above)
Comment (nhatminh-96) [reply crawl_tool/crawl.py:11]: I think we run this script one time and mainly on our cloud instance and files will be stored at that address. But yes output_folder option argument has been added.

**Comment by thinhngo-x** [comment]
@Viet1004 Merging is still blocked. Pls review branch protection rules.

**Review by Viet1004 — APPROVED** [review]

**Pull request merged**